### PR TITLE
Introduce `max_conns` for DatabaseConfig

### LINKED ISSUE #22: Sqlite "Database is Locked" Error
In running some builds that reference many crates, against a Alexandrie+Sqlite setup, I started getting a 500 error from cargo, where Alexandrie logs showed:

```
SqlError("Database is locked")
```

To reproduce:
1) Stand up API using sqlite
2) Publish 4-5 crates
3) Have local project that depends on those 4-5 crates 
4) Repeatedly run build from clean environment (reset $CARGO_HOME each time)

It seems sqlite doesn't allow multiple writes at the same time.
To address it, I had to set sqlite to a WriteAheadLog mode:
```
sqlite3 alexandrie.db 'PRAGMA journal_mode=WAL;'
```
and limit the number of connections in the connection pool to 1

```rust
let mut builder = r2d2::Builder::default();
builder = builder.max_size(1);
```
(seemingly, because still only 1 connection is allowed to hold onto the corresponding `alexandrie.wal` file that gets created)

I modified `DatabaseConfig` in this [commit](https://github.com/agoops/alexandrie/commit/809fa4541850975580a227a78e5a507d8bfbe067) to accept an optional `connection_pool_max_size`. These 2 fixes, and restarting the app, solved the "Database is Locked" problem I was seeing.

Let me know what you think, or other suggestions to fix this or extend the Database configuration options to accommodate sqlite. I can open a PR for this if you think this is an appropriate change.

### PR DESCRIPTION
**Pull request by agoops**
As noted in this [issue](https://github.com/Hirevo/alexandrie/issues/22), sqlite requires 1 connection open when set to "WAL" (WriteAheadLog) mode. 

`diesel-rs` defaults to a connection pool [size of 10](https://docs.diesel.rs/diesel/r2d2/struct.Builder.html#method.max_size), so this PR introduces an optional config value that can be used to set that value

**Closes #22.**

### CODE REVIEW

**Comment by Hirevo** [comment]
Thanks for the PR, I'll be reviewing it a little bit later today, when I get home.

File: src/config.rs [hunk 1]
```diff
@@ -54,6 +54,8 @@ pub struct FrontendConfig {
 pub struct DatabaseConfig {
     /// The database connection URL.
     pub url: String,
+    /// Max connection pool size
+    pub connection_pool_max_size: Option<u32>,
```
Comment (Hirevo) [inline src/config.rs]: I would suggest having a simpler name for this configuration option, something like:
```suggestion
    pub max_conns: Option<u32>,
```

File: README.md [hunk 2]
```diff
@@ -65,6 +65,13 @@ url = "<path-to-sqlite-file>"
 url = ":memory:" # ephemeral in-memory database, doesn't persists between restarts
 ```
 
+Optionally, specify the maximum number of threads for the database connection pool. 
+
+```toml
+connection_pool_max_size = 1
```
Comment (Hirevo) [inline README.md]: Just to reflect the suggested change in `src/config.rs`.
```suggestion
max_conns = 1
```

File: src/db/mod.rs [hunk 3]
```diff
@@ -48,9 +49,13 @@ impl<T> Repo<T>
 where
     T: diesel::Connection + 'static,
 {
-    /// Constructs a `Repo<T>` for the given database URL (creates a connection pool).
-    pub fn new(database_url: &str) -> Self {
-        Self::from_pool_builder(database_url, r2d2::Builder::default())
+    /// Constructs a `Repo<T>` for the given database config (creates a connection pool).
+    pub fn new(database_config: &DatabaseConfig) -> Self {
+        let mut builder = r2d2::Builder::default();
+        if let Some(max_size) = database_config.connection_pool_max_size {
```
Comment (Hirevo) [inline src/db/mod.rs]: Just to reflect the suggested change in `src/config.rs`.
```suggestion
        if let Some(max_size) = database_config.max_conns {
```

File: src/config.rs (hunk 1 above)
Comment (agoops) [reply src/config.rs]: Good call! Committing..

**Comment by agoops** [comment]
@Hirevo , thanks for taking the time to review. Please merge whenever!

**Comment by Hirevo** [comment]
Thank you again for the PR !
I am merging right now. 🎉

**Pull request merged**